core/src/mast/serialization/basic_block_data_builder.rs

```diff
@@ -37,12 +37,14 @@ impl BasicBlockDataBuilder {
 /// Mutators
 impl BasicBlockDataBuilder {
     /// Encodes a [`BasicBlockNode`] into the serialized [`crate::mast::MastForest`] data field.
-    pub fn encode_basic_block(&mut self, basic_block: &BasicBlockNode) {
+    pub fn encode_basic_block(&mut self, basic_block: &BasicBlockNode, mast_forest: &MastForest) {
         // 2nd part of `mast_node_to_info()` (inside the match)
         for op_or_decorator in basic_block.iter() {
             match op_or_decorator {
                 OperationOrDecorator::Operation(operation) => operation.write_into(&mut self.data),
-                OperationOrDecorator::Decorator(decorator) => self.encode_decorator(decorator),
+                OperationOrDecorator::Decorator(&decorator_id) => {
+                    self.encode_decorator(&mast_forest[decorator_id])
+                },
```

Comment:
>  For some decorators this may be fine (as they are smaller than 32 bits), but for things AsmOp (which we currently use for debug info) this would be quite expensive.

In either strategy, we'll need to serialize the `AsmOp` at least once. So we gain by storing each `AsmOp` once, and then encoding the `DecoratorId` in basic blocks only if that specific `AsmOp` is used more than once. But all `AsmOp`s are going to be unique, right?

So IIUC, we end up encoding *more* bytes using your suggestion, since we need, for each `AsmOp`
1. however many bytes for that `AsmOp`
2. an extra 4 bytes for the `DecoratorId`

as opposed to storing the `AsmOp` directly and only storing those bytes (i.e. saving 4 bytes per `AsmOp` used).

Does that sound right?

Reply:
Addressed in #1482